yt_dlp/YoutubeDL.py

```diff
@@ -3560,14 +3560,9 @@ def download(self, url_list):
 
         return self._download_retcode
 
-    def download_with_info_file(self, info_filename):
-        with contextlib.closing(fileinput.FileInput(
-                [info_filename], mode='r',
-                openhook=fileinput.hook_encoded('utf-8'))) as f:
-            # FileInput doesn't have a read method, we can't call json.load
-            infos = [self.sanitize_info(info, self.params.get('clean_infojson', True))
-                     for info in variadic(json.loads('\n'.join(f)))]
-        for info in infos:
+    def download_with_info(self, info_list):
+        """Download using already extracted info_dicts."""
+        for info in info_list:
             try:
                 self.__download_wrapper(self.process_ie_result)(info, download=True)
```

Comment:
It would be more useful to yield these processed `info_dict` than discard them
```suggestion
                yield self.__download_wrapper(self.process_ie_result)(info, download=True)
```
and then adding a `return res` to `__download_wrapper`

`download_with_info_file` can discard them to retain old behavior

Reply:
Done!

Reply:
> It would be more useful to yield these processed `info_dict` than discard them
> 
> and then adding a `return res` to `__download_wrapper`
> 
> `download_with_info_file` can discard them to retain old behavior

I didn't want to alter any public API but if that's what you think should be done i will implement it